python-package/lightgbm/plotting.py

```diff
@@ -414,13 +415,38 @@ def plot_metric(
     return ax
 
 
+def _determine_direction_for_numeric_split(
+    fval: float,
+    threshold: float,
+    missing_type_str: str,
+    default_left: bool,
+) -> str:
+    missing_type = _MissingType(missing_type_str)
+    if math.isnan(fval) and missing_type != _MissingType.NAN:
+        fval = 0.0
+    if ((missing_type == _MissingType.ZERO and _is_zero(fval))
+            or (missing_type == _MissingType.NAN and math.isnan(fval))):
+        direction = 'left' if default_left else 'right'
+    else:
+        direction = 'left' if fval <= threshold else 'right'
+    return direction
+
+
+def _determine_direction_for_categorical_split(fval: float, thresholds: str) -> str:
+    if math.isnan(fval):
+        return 'right'
```

Comment:
https://github.com/microsoft/LightGBM/blob/f94050a4cc94909e10a0064baff11cec795eb250/include/LightGBM/tree.h#L374-L381
```suggestion
    if math.isnan(fval) or int(fval) < 0:
        return 'right'
```